File: 1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.java
 class Solution {
        public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
            if (original == null) {
                return null;
            }
            if (original.val == target.val) {
                return cloned;
            }
            TreeNode left = getTargetCopy(original.left, cloned.left, target);
            if (left != null && left.val == target.val) {
                return left;
            }
            return getTargetCopy(original.right, cloned.right, target);
        }
    }

File: 1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree/1498-find-a-corresponding-node-of-a-binary-tree-in-a-clone-of-that-tree.py
class Solution:
    def getTargetCopy(
        self, original: TreeNode, cloned: TreeNode, target: TreeNode
    ) -> TreeNode:
        res = None

        def dfs(original, cloned):
            nonlocal res
            if cloned is None:
                return
            if original == target:
                res = cloned
                return
            dfs(original.left, cloned.left)
            dfs(original.right, cloned.right)

        dfs(original, cloned)
        return res